### PR TITLE
Add an indicator for the current time in TimetableGrid

### LINKED ISSUE #537: Add a line to TimetableGrid to indicate the current session
Could we implement a feature to display the current time session?
It would be beneficial to have a screenshot test or preview for this feature.
https://github.com/DroidKaigi/conference-app-2024/blob/main/feature/sessions/src/androidUnitTest/kotlin/io/github/droidkaigi/confsched/sessions/TimetableScreenTest.kt

### PR DESCRIPTION
**Pull request by kitakkun**
## Issue
- close #537 

## Overview (Required)
- Add an indicator for the current time in TimetableGrid and TimetableGridHours.
- Add tests to verify the TimeLine is displayed as expected.

## Additional Notes
To achieve the desired visual result, the indicator must be drawn in both TimetableGrid and TimetableGridHours.
- Both have a `clipToBounds` modifier, which causes the circular part of the indicator to be cut off.
- If we draw the indicator before applying the `clipToBounds` modifier, it could be drawn behind the tabs, which is also an undesired result.

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :--:
 | 

## Movie (Optional)
<video src="https://github.com/user-attachments/assets/e187d7a5-9250-4fe0-9cea-6f68a8eeacdc" width="300" >

### CODE REVIEW

**Comment by charles-b-stb** [comment]
@kitakkun If you can, please add screenshots of before and after your commit as it helps reviewing the pull request.

File: feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/TimetableScreenPresenter.kt [hunk 1]
```diff
@@ -62,6 +73,11 @@ fun timetableScreenPresenter(
             }
         }
     }
+
+    LifecycleEventEffect(Lifecycle.Event.ON_RESUME) {
+        timeLine = TimeLine.now(clock)
+    }
```
Comment (kitakkun) [inline feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/TimetableScreenPresenter.kt:79]: After the app returns to the foreground, we should update the `timeLine` to reflect the latest state.

File: feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableGridHours.kt [hunk 2]
```diff
@@ -96,6 +106,16 @@ fun TimetableGridHours(
                     linePxSize,
                 )
             }
+            .drawWithContent {
```
Comment (takahirom) [inline feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/component/TimetableGridHours.kt:109]: 👍

**Review by takahirom — APPROVED** [review]
Looks great! Well-structured code! Thank you for your contribution!

**Pull request merged**